docs/adr/0019-use-1password-for-sharing-secrets.md

```diff
@@ -0,0 +1,96 @@
+  1. There should be a central source of truth for the secrets.
+  1. The scheme for storing, sharing, and retrieving secrets should be
+     consistent with our understanding of best security practices.
+  1. The scheme should also be user-friendly and accessible without assuming
+     any knowledge of back-end infrastructure or overly complex authentication
+     mechanisms.
+
+## Considered Alternatives
+
+* Email
+* Slack DMs
+* 1Password
+* AWS SSM / ParameterStore
+
+## Decision Outcome
+
+Chosen Alternative: Use a 1Password vault for sharing secrets.
+
+* 1Password allows us to control access to the secrets repository on a per-user
+  basis, and updates to any secret will be available to other users on the
+  vault immediately.  With a separate vault within 1Password, we need not share
+  secrets with other Truss personnel who are not assigned to the project.
+* 1Password also has familiar and user-friendly graphical interfaces via the
+  web and a multi-platform desktop application.
+* Access to secrets can be revoked as quickly and as easily as it was granted.
+
+Consequences:
+
+* We will need to create a vault, which takes time and costs money.
+* Some synchronization will be required on how to use the vault and what
+  secrets to store in it.
+* Secrets that are required by the easi-backend app will have to be updated in
+  SSM anyway, which means changing them requires updating the value in two
+  places (1Password and SSM).
+
+## Pros and Cons of the Alternatives
+
+### Email
+
+* `+` Fast.
+* `+` Some of our peers share secrets with us using email, so forwarding the
+  emails is an easy and straightforward way to distribute secrets.
+* `-` Poor access controls: once an email is sent, its contents cannot be recalled.
+* `-` Emails can be easily shared with a wide audience inadvertently, resulting
+  in a pervasive risk of uncontrolled disclosure.
+* `-` If we have to manage any secrets that cannot be rotated, there will be no
+  way to guarantee that sent emails are burned after reading: rather, they will
+  likely live on in the archived email of whoever received them.
+* `-` There would be no central source of truth on what the current secrets
+  are, which complicates onboarding/off-boarding, engineers returning from
+  vacation, etc.
+
+### Slack DMs
+
+* `+` It's easy to find people in Slack, and we all use it.
+* `+` Only two people are in a DM thread. Unlike email, it is not easy to share
+  contents of a DM inadvertently.
+* `-` Due to CMS record-keeping policies, DMs cannot be deleted and may be
+  subject to subpoena or FOIA requests, which makes them public records.
+* `-` There is no central source of truth.
+
+### 1Password
+
+* `+` Per-user access controls.
+* `+` We can restrict the vault to project engineers according to the principle
+  of least access.
+* `+` The 1Password vault is a central source of truth on secrets.
+* `-` We have to create a vault, which costs money. May require CORS approval.
```

Comment:
```suggestion
* `-` We have to create a vault, which costs money.
```

Reply:
Doesn't it?

Reply:
There are different alternatives here. Is the main point that there is a cost involved?

Reply:
I'm not sure what CORS approval is: is that about approving the expense or the use of the tech? If it's about the tech, that sounds like a separate bullet.

In terms of the cost (ignoring approval), I think that only matters if the amount moves the budget needle. If you need to create a new 1Password Teams account then it's ~$50/person/yr.

Reply:
2b594ca7778142a434469e78db09674326d7e7cb

Reply:
I think that if there are bureaucratic hurdles to getting a 1Password vault, then that is going to be a bigger cost than the subscription, yes.